### PR TITLE
Pull based config via credstash and localfile

### PR DESCRIPTION
**Pull request by elbuo8**

### CODE REVIEW

File: bin/limitd [hunk 1]
```diff
@@ -3,11 +3,31 @@
...
+
 const LimitdServer = require('../server');
 const server = new LimitdServer(config);
 
 server.start();
 
+server.on('started', () => {
+  if (config.configurator) {
+    config.configurator.metrics = metrics;
+    config.configurator.logger = logger;
```
Comment (dirceu) [inline bin/limitd]: Any reason not to pass the agent itself?

Also, the instrumentation agent is a singleton so we can just require auth0-instrumentation again and use it anywhere (if it was initialized before).

File: server.js [hunk 2]
```diff
@@ -168,21 +163,22 @@ LimitdServer.prototype.stop = function (callback) {
...
-      self.emit('close');
+      this.emit('close');
       return callback(serverCloseError || dbCloseError);
     });
   });
-
 };
 
+LimitdServer.prototype.updateBucketConfig = function (buckets) {
+  this._db.loadTypes(buckets);
```
Comment (dirceu) [inline server.js]: Have you tested this under load to see how the server behaves? Do we fail all requests while this reload, or we wait...?

Comment (elbuo8) [reply server.js]: Expected behavior is we wait.

File: lib/pipeline/RequestHandler.js [hunk 3]
```diff
@@ -105,14 +112,12 @@ class RequestHandler extends Duplex {
...
+      var queued = this.queue.shift();
+      if (!this.push(queued)) {
+        break;
+      }
     }
-
-    this.queue = i === this.queue.length ? [] : this.queue.slice(i);
   }
 
   _read() {
```
Comment (jfromaniello) [inline lib/pipeline/RequestHandler.js]: `this.reading` is not longer needed when using the while loop

**Pull request closed without merging**